services/web/server/src/simcore_service_webserver/projects/db.py

```diff
@@ -552,124 +406,213 @@ async def list_projects_full_search(
-                    & (projects.c.type == ProjectType.STANDARD)
-                    & (
-                        (projects.c.name.ilike(f"%{text}%"))
-                        | (projects.c.description.ilike(f"%{text}%"))
-                        | (projects.c.uuid.ilike(f"%{text}%"))
+                        projects.c.trashed_at.is_not(None)
+                        & projects.c.trashed_explicitly.is_(True)
                     )
+                    if filter_trashed
+                    # not marked as trashed
+                    else projects.c.trashed_at.is_(None)
                 )
-            )
-
-            if tag_ids_list:
-                shared_workspace_query = shared_workspace_query.where(
-                    sa.func.coalesce(
-                        project_tags_subquery.c.tags,
-                        sa.cast(sa.text("'{}'"), sa.ARRAY(sa.Integer)),
-                    ).op("@>")(tag_ids_list)
+            if filter_by_text is not None:
+                attributes_filters.append(
+                    (projects.c.name.ilike(f"%{filter_by_text}%"))
+                    | (projects.c.description.ilike(f"%{filter_by_text}%"))
+                    | (projects.c.uuid.ilike(f"%{filter_by_text}%"))
+                    | (users.c.name.ilike(f"%{filter_by_text}%"))
                 )
 
-            combined_query = sa.union_all(
-                private_workspace_query, shared_workspace_query
-            )
+            if folder_query.folder_scope is not FolderScope.ALL:
+                if folder_query.folder_scope == FolderScope.SPECIFIC:
+                    attributes_filters.append(
+                        projects_to_folders.c.folder_id == folder_query.folder_id
+                    )
+                else:
+                    assert folder_query.folder_scope == FolderScope.ROOT  # nosec
+                    attributes_filters.append(projects_to_folders.c.folder_id.is_(None))
+
+            ###
+            # Combined
+            ###
+
+            combined_query = None
+            if (
+                private_workspace_query is not None
+                and shared_workspace_query is not None
+            ):
+                combined_query = sa.union_all(
+                    private_workspace_query.where(sa.and_(*attributes_filters)),
+                    shared_workspace_query.where(sa.and_(*attributes_filters)),
+                )
+            elif private_workspace_query is not None:
+                combined_query = private_workspace_query.where(
+                    sa.and_(*attributes_filters)
+                )
+            elif shared_workspace_query is not None:
+                combined_query = shared_workspace_query.where(
+                    sa.and_(*attributes_filters)
+                )
 
-            count_query = sa.select(func.count()).select_from(combined_query)
+            if combined_query is None:
+                msg = "No valid queries were provided to combine."
```

Comment:
You might want to enhance this message and point out to the combination of inputs to this function that could have triggered this state

Reply:
improved